### PR TITLE
feat(content-sharing): Invite collaborators

### PR DESCRIPTION
**Pull request by sahiga**
Invite collaborators in the ContentSharing Element.

content-sharing-invite-collabs

Note: The Invite Collaborators Modal will remain open after the POST request to the Collaborations API. This behavior will be modified in a follow-up PR.

### CODE REVIEW

File: src/features/unified-share-modal/utils/convertData.js [hunk 1]
```diff
@@ -263,37 +267,80 @@ export const convertCollabsResponse = (
...
+    entries.forEach(collab => {
+        // Only show accepted collaborations
+        if (collab.status && collab.status === STATUS_ACCEPTED) {
+            const {
+                accessible_by: { id: userID, login: email, name, type },
+                id: collabID,
+                expires_at: executeAt,
+                role,
+            } = collab;
+            const collabEmailDomain = email ? email.split('@')[1] : null;
```
Comment (booc0mtaco) [inline src/features/unified-share-modal/utils/convertData.js:281]: i see some email parsing logic in this and nearby lines. should this be collected into 1-2 utility functions?

File: src/common/types/core.js [hunk 2]
```diff
@@ -412,16 +412,26 @@ type Collaborators = {
...
 };
 
-type Collaboration = {
-    accessible_by: {
-        id: number | string,
-        login: string,
-        name: string,
-        type: 'user',
-    },
+type AccessibleBy = {
```
Comment (booc0mtaco) [inline src/common/types/core.js:415]: this type looks familiar. Does this exist already somewhere? `UserMini`? If not, maybe a name like `AccessibleByUser` will explain its provenance a bit more clearly.

File: src/elements/content-sharing/ContentSharing.js [hunk 3]
```diff
@@ -156,72 +158,75 @@ function ContentSharing({ apiHost, displayInModal, itemID, itemType, language, t
...
-        const { accessLevel = '', serverURL } = sharedLink;
-        return (
-            <Internationalize language={language} messages={usmMessages}>
-                <>
-                    <SharingNotification
-                        accessLevel={accessLevel}
-                        api={api}
+    if (!item || !sharedLink || !currentUserID || !sharedLink.serverURL) {
+        return null;
+    }
```
Comment (manishPh) [inline src/elements/content-sharing/ContentSharing.js:163]: 👍

File: src/features/unified-share-modal/utils/convertData.js (hunk 1 above)
Comment (manishPh) [reply src/features/unified-share-modal/utils/convertData.js:281]: Some email validating/parsing functions here: https://github.com/box/box-ui-elements/blob/master/src/utils/parseEmails.js & https://github.com/box/box-ui-elements/blob/master/src/utils/validators.js

File: src/features/unified-share-modal/utils/convertData.js [hunk 4]
```diff
@@ -263,37 +267,80 @@ export const convertCollabsResponse = (
...
-            name,
-            translatedRole: `${role[0].toUpperCase()}${role.slice(1)}`, // capitalize the user's role
-            type,
-            userID: parseInt(userID, 10),
-        };
-        if (executeAt) {
-            convertedCollab.expiration = { executeAt };
+
+    const collaborators = [];
+    entries.forEach(collab => {
```
Comment (manishPh) [inline src/features/unified-share-modal/utils/convertData.js:272]: Minor: Could do
```
const { entries = [] } = collabsAPIData;
const collaborators = entries.filter(collab.status === STATUS_ACCEPTED).map(...);
return { collaborators };
```

**Comment by sahiga** [comment]
@booc0mtaco @swfree @manishPh Thanks for reviewing 🙂 I updated the code based on your comments. Could you take another look?

**Pull request merged**